rfcs/45-stageless.md

```diff
@@ -0,0 +1,980 @@
+                apply_system_buffers.named(MySystems::FlushProjectiles)
+                fork_projectiles,
+            ]
+            .after(CoreSystems::FlushPostUpdate)
+            .before(CoreSystems::FlushLast)
+        )
+    }
+}
+```
+
+### Running other systems on-demand with exclusive systems
+
+Exclusive systems have other uses as well.
+
+All systems and system sets added to an `App` are stored within a resource called `Systems`.
+If you want to run a system or system set, you have to extract it from `Systems` first.
+(This lets `Systems` remain accessible when those systems are running).
+
+`Systems` has methods to extract a system or system set given its label.
+When you extract a system set, you actually receive a **schedule**.
+This is the executable version of the set, containing all its systems and conditions, along with instructions to run them in the correct order.
+After running a schedule, you can return it to `Systems`.
+A `World::schedule_scope` method is provided for this common "extract-run-return" case.
+
+The default `App` runner itself actually uses these methods to run the startup sequence and main update loop, so if you retrieve and run a single system somewhere, you'll effectively have written a very basic executor!
+
+```rust
+fn fancy_exclusive_system(world: &mut World) {
+    // removes schedule from Systems, executes fn with it, then returns it to Systems
+    world.schedule_scope(ExampleLabel, |world, mut schedule| {
+        schedule.run(world);
+    });
+}
+```
+
+This pattern is your go-to when your scheduling needs grow beyond "each system runs once per app update".
+You might want to:
+
+- repeatedly loop over a sequence of game logic several times in a single app update (e.g. fixed timestep, action queue)
+- have complex branches in your schedule (e.g. state transitions)
+- change to a different threading model for some portion of your systems
+- integrate an external service with arbitrary side effects, like scripting or a web server
+
+As long as you have `&mut World`, from an exclusive system or command, you can extract anything available in the `Systems` resource and run it.
+
+Unlike in previous versions of Bevy, states and the fixed timestep are no longer powered by run criteria.
+Instead, systems that are part of states or fixed time steps are simply added under a separate system set that will be retrieved and ran by an exclusive system.
+As a result, you can transition states inside the fixed timestep without issue.
+
+### Fixed timestep
+
+A **fixed timestep** advances a fixed number of times each second.
+
+It's implemented as an exclusive system that runs a schedule zero or more times in a row (depending on how long the previous app update took to complete).
+When you supply a constant delta time value (the literal fixed timestep) inside the encapsulated systems, the result is consistent and repeatable behavior regardless of framerate (or even the presence of a GPU).

```

Comment:
I think a small code example on how to add a system that runs in fixed time steps would be nice here (similar to the states one below)

Reply:
Sure. It is pretty much the same as states tho.

You'll add things under a specific system set (i.e. `CoreSet::FixedUpdate`). And then an exclusive system (probably added by `bevy_time`) would just load and run that schedule.